build.gradle

```diff
@@ -190,15 +197,24 @@ subprojects {
 			}
 		}
 	}
-	signing {
-		sign publishing.publications.mavenJava
-	}
-	publishMavenJavaPublicationToMavenRepository.doFirst {
-		if(OSSRH_URL == null) {
-			throw new GradleException("Please specify the variable 'ossrhUrl' in '~/.gradle/gradle.properties'.")
-		}
-		if(version == "unspecified") {
-			throw new GradleException("Please specify the version to release with '-Pversion=X.Y.Z'.")
+
+	publish.doFirst {
+		if(OSSRH_TARGET == 'public') {
+			signing {
+					sign publishing.publications.mavenJava
+			}
+			publishMavenJavaPublicationToMavenRepository.doFirst {
+				if(OSSRH_URL == null) {
+					throw new GradleException("Please specify the variable 'ossrhUrl' in '~/.gradle/gradle.properties'.")
+				}
+				if(version == "unspecified") {
+					throw new GradleException("Please specify the version to release with '-Pversion=X.Y.Z'.")
+				}
+			}
+		} else if(OSSRH_TARGET == 'local') {
+			// nothing
+		} else {
+			throw new GradleException("When 'publish' is invoked, '-PossrhTarget=[public|local]' must be specified.")
```

Comment:
sidenote: we should consider renaming all `ossrh` to `publish` some day

Reply:
I think so, too